content/en/blog/_posts/2021-03-09-Defining-NetworkPolicy-Conformance-For-CNIs

```diff
@@ -0,0 +1,473 @@
+  ability to target a named port; others don't work with port ranges).
+
+Thus, end-users need to follow a multistep process to implement Network Policies to secure their applications:
+- Confirm that their network plugin supports NetworkPolicies (some don't, such as Flannel)
+- Confirm that their cluster's network plugin supports the specific NetworkPolicy features that they are interested in (again, the named port or port range examples come to mind here)
+- Confirm that their application's Network Policy definitions are doing the right thing
+- Find out the vendor specific implementation of a policy, and check whether or not that implementation had a CNI neutral implementation (which is preferable for most users)
+
+The NetworkPolicy project in upstream Kubernetes aims at providing a community where
+people can learn about, and contribute to, the Kubernetes NetworkPolicy API and the surrounding ecosystem.
+
+## The First step: A validation framework for NetworkPolicies that was intuitive to use and understand
+
+The Kubernetes end to end suite has always had NetworkPolicy tests, but these weren't
+run in CI, and the way they were implemented didn't provide holistic, easily consumable
+information about how a policy was working in a cluster.
+This is because the original tests didn't provide any kind of visual summary of connectivity
+across a cluster.   We thus initially set out to make it easy to confirm CNI support for NetworkPolicies by
+making the end to end tests (which are often used by administrators or users to diagnose cluster conformance) easy to interpret.
+
+To solve the problem of confirming that CNIs support the basic features one cares about
+for a policy, we built a new NetworkPolicy validation tool into the Kubernetes e2e
+framework which allows for visual inspection of policies and their effect on a standard set of pods in a cluster.
+For example, take the following test output.  We found a bug in
+[ovn kubernetes](https://github.com/ovn-org/ovn-kubernetes/issues/1782) with this tool
+that was really easy to characterize, wherein certain policies caused a state-modification that,
+later on, caused traffic to incorrectly be blocked (even after all Network Policies, cluster wide, were deleted).
+
+This is the network policy for the test in question:
+```yaml
+Nov  3 15:58:54.034: INFO: Network Policy creating x/allow-ingress-port-80
+metadata:
+  creationTimestamp: null
+  name: allow-ingress-port-80
+spec:
+  ingress:
+  - ports:
+    - port: serve-80-tcp
+  podSelector: {}
+```
+
+These are the expected connectivity results.  The test setup is 9 pods (3 namespaces: x, y, and z;
+and 3 pods in each namespace of a, b, and c); each pod runs a server on the same port and protocol
+that can be reached through HTTP calls in the absence of network policies.  Connectivity is verified
+by using the agnhost network utility to issue HTTP calls on a port and protocol that other pods are
```

Comment:
I guess it would be good to put a link here to agnhost so people can understand what it is: 

```suggestion
by using the [agnhost](https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost) network utility to issue HTTP calls on a port and protocol that other pods are
```